python/src/wirecraft_server/server.py

```diff
@@ -23,122 +26,134 @@ class Server:
     """
 
     def __init__(self) -> None:
-        self.client_connexions: list[Any] = []  # TODO
+        self.client_connexions: list[Any] = []  # TODO(airopi): WS
         self._last_refresh: float = time.perf_counter()
+        self._stop = asyncio.Event()
 
-    def _wait_next_refresh(self):
+    def start(self):
+        logger.info("Server started!")
+        try:
+            asyncio.run(self._run())
+        except KeyboardInterrupt:
+            logger.info("Server stopped!")
+            self._stop.set()
+        except Exception:
+            logger.exception("Server crashed!")
+            raise SystemExit(1)
+
+    async def _wait_next_refresh(self):
         delay = 1 / REFRESH_RATE
         _next = self._last_refresh + delay
         _now = time.perf_counter()
         _wait = _next - _now
         if _wait < 0:
-            # TODO: implement server slow down (increase delay in case of poor performances)
+            # TODO(airopi): implement server slow down (increase delay in case of poor performances)
             logger.warning("Can't keep up ! %ss behind the normal refresh", -round(_wait, 3))
             self._last_refresh = _now
+            return self._stop.is_set()
```

Comment:
Shouldn't this be removed?

Reply:
Yes and no.
In the current implementation, only a CTRL+C can stop the server, so it emit a KeyboardInterrupt exception that is handled in the start() method, and so setting `self._stop.set()` don't have any impact.

But, two things could make this usefull:
- adding any new way to stop the server (ie a /stop command (not planned though))
- if the server start any kind of "subtask", we could use task.cancel(), or make the task look the the _stop event to see if it is set or not

So, you are right : we will most likely never use this, but if one day we *need* it, it will be present 👍

Reply:
Understood, I'd suggest adding a comment to explain in case we (I) forget later